manifests/init.pp

```diff
@@ -1,269 +1,277 @@
-  Boolean              $stunnel                = simplib::lookup('simp_options::stunnel', { 'default_value' => false }),
-  Boolean              $stunnel_tcp_nodelay    = true,
-  Array[String]        $stunnel_socket_options = [],
-  Boolean              $stunnel_systemd_deps   = true,
-  Array[String]        $stunnel_wantedby       = []
+  Boolean               $is_server                     = false,
+  Boolean               $is_client                     = true,
+  Boolean               $nfsv3                         = false,
+  Boolean               $gssd_avoid_dns                = true,
+  Boolean               $gssd_limit_to_legacy_enctypes = false,
+  Boolean               $gssd_use_gss_proxy            = true,
+  Simplib::Port         $lockd_port                    = 32803,
+  Simplib::Port         $lockd_udp_port                = 32769,
+  Simplib::Port         $nfsd_port                     = 2049,
+  Simplib::Port         $sm_notify_outgoing_port       = 2021,
+  Simplib::Port         $statd_port                    = 662,
+  Simplib::Port         $statd_outgoing_port           = 2020,
+  Nfs::NfsConfHash      $custom_nfs_conf_opts          = {},
+  Nfs::LegacyDaemonArgs $custom_daemon_args            = {},
+  Boolean               $idmapd                        = false,
+  Boolean               $secure_nfs                    = false,
+  Boolean               $ensure_latest_lvm2            = true,
+  Boolean               $kerberos                      = simplib::lookup('simp_options::kerberos', { 'default_value' => false }),
+  Boolean               $keytab_on_puppet              = simplib::lookup('simp_options::kerberos', { 'default_value' => true}),
+  Boolean               $firewall                      = simplib::lookup('simp_options::firewall', { 'default_value' => false}),
+  Boolean               $stunnel                       = simplib::lookup('simp_options::stunnel', { 'default_value' => false }),
+  Simplib::Port         $stunnel_nfsd_port             = 20490,
+  Array[String]         $stunnel_socket_options        = ['l:TCP_NODELAY=1','r:TCP_NODELAY=1'],
+  Integer               $stunnel_verify                = 2,
+  Boolean               $tcpwrappers                   = simplib::lookup('simp_options::tcpwrappers', { 'default_value' => false }),
+  Simplib::Netlist      $trusted_nets                  = simplib::lookup('simp_options::trusted_nets', { 'default_value' => ['127.0.0.1'] })
 ) {
 
   simplib::assert_metadata($module_name)
-
-  if $stunnel_tcp_nodelay {
-    $_stunnel_socket_options = $stunnel_socket_options + [
-      'l:TCP_NODELAY=1',
-      'r:TCP_NODELAY=1'
-    ]
-  }
-  else {
-    $_stunnel_socket_options = $stunnel_socket_options
+  if (versioncmp($facts['os']['release']['full'], '7.4') < 0) {
+    warning("This version of simp-nfs may not work with ${facts['os']['name']} ${facts['os']['release']['full']}. Use simp-nfs module version < 7.0.0 instead")
```

Comment:
Do we want to add something like this capability to `simplib::assert_metadata`? No need to change this, just wanted to see if it would be interesting.